diff -Naur src/blosc/CMakeLists.txt external_blosc/blosc/CMakeLists.txt
--- src/blosc/CMakeLists.txt	2016-02-03 10:26:28 -0700
+++ external_blosc/blosc/CMakeLists.txt	2017-03-03 09:03:31 -0700
@@ -61,6 +61,8 @@
         set(SOURCES ${SOURCES} win32/pthread.c)
     else(NOT Threads_FOUND)
         set(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
+        set(LIBS ${LIBS} ${PTHREAD_LIBS})
+        include_directories( ${PTHREAD_INCLUDE_DIR} )
     endif(NOT Threads_FOUND)
 else(WIN32)
     find_package(Threads REQUIRED)
diff -Naur external_blosc.orig/blosc/blosc.c external_blosc/blosc/blosc.c
--- external_blosc.orig/blosc/blosc.c   2018-07-30 04:56:38 -0600
+++ external_blosc/blosc/blosc.c        2018-08-11 15:27:26 -0600
@@ -56,14 +56,7 @@
   #include <inttypes.h>
 #endif  /* _WIN32 */

-/* Include the win32/pthread.h library for all the Windows builds. See #224. */
-#if defined(_WIN32)
-  #include "win32/pthread.h"
-  #include "win32/pthread.c"
-#else
-  #include <pthread.h>
-#endif
-
+#include <pthread.h>

 /* Some useful units */
 #define KB 1024
