--- _msvccompiler.py.orig	2017-01-17 00:57:48 -0700
+++ _msvccompiler.py	2017-05-20 09:47:26 -0600
@@ -237,7 +237,7 @@
             ldflags.extend(('/nodefaultlib:libucrt.lib', 'ucrt.lib'))
 
         ldflags_debug = [
-            '/nologo', '/INCREMENTAL:NO', '/LTCG', '/DEBUG:FULL'
+            '/nologo', '/INCREMENTAL:NO', '/LTCG'
         ]
 
         self.ldflags_exe = [*ldflags, '/MANIFEST:EMBED,ID=1']
