diff -Naur orig/common/sys/platform.h external_embree/common/sys/platform.h
--- orig/common/sys/platform.h	2020-05-13 23:08:53 -0600
+++ external_embree/common/sys/platform.h	2020-06-13 17:40:26 -0600
@@ -84,8 +84,8 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #ifdef __WIN32__
-#define dll_export __declspec(dllexport)
-#define dll_import __declspec(dllimport)
+#define dll_export 
+#define dll_import 
 #else
 #define dll_export __attribute__ ((visibility ("default")))
 #define dll_import 
diff --git orig/common/tasking/CMakeLists.txt external_embree/common/tasking/CMakeLists.txt
--- orig/common/tasking/CMakeLists.txt
+++ external_embree/common/tasking/CMakeLists.txt
@@ -27,7 +27,11 @@
     else()
       # If not found try getting older TBB via module (FindTBB.cmake)
       unset(TBB_DIR CACHE)
-      find_package(TBB 4.1 REQUIRED tbb)
+      if (TBB_STATIC_LIB)
+        find_package(TBB 4.1 REQUIRED tbb_static)
+      else()
+        find_package(TBB 4.1 REQUIRED tbb)
+      endif()
       if (TBB_FOUND)
         TARGET_LINK_LIBRARIES(tasking PUBLIC TBB)
         TARGET_INCLUDE_DIRECTORIES(tasking PUBLIC "${TBB_INCLUDE_DIRS}")
