--- hidapi/hidapi.h	2011-10-25 20:58:16 -0600
+++ hidapi/hidapi.h	2016-11-01 12:05:58 -0600
@@ -30,7 +30,11 @@
 #include <wchar.h>
 
 #ifdef _WIN32
-      #define HID_API_EXPORT __declspec(dllexport)
+      #ifdef HID_API_STATIC
+        #define HID_API_EXPORT 
+      #else
+        #define HID_API_EXPORT __declspec(dllexport)
+      #endif
       #define HID_API_CALL
 #else
       #define HID_API_EXPORT /**< API export macro */
