diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50eb25e..b13d2da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -585,12 +585,16 @@ install(FILES ${HIPRT_ORO_HEADERS}
 		DESTINATION include/contrib/Orochi/ParallelPrimitives)
 
 # add hipfb files
-file(GLOB HIPRT_HIPFB_FILES "${BASE_OUTPUT_DIR}/${CMAKE_BUILD_TYPE}/*.hipfb")
-install(FILES ${HIPRT_HIPFB_FILES}
-		DESTINATION bin)
-
-
+if(PRECOMPILE)
+	install(FILES ${KERNEL_HIPRT_COMP} ${KERNEL_OROCHI_COMP}
+			DESTINATION bin)
+endif()
 
+# Baked headers.
+if(BAKE_COMPILED_KERNEL)
+	install(FILES ${KERNEL_HIPRT_H}
+			DESTINATION include/hiprt/impl)
+endif()
 
 # Project: Unit Test
 if(NOT NO_UNITTEST)
