diff -Naur external_igc_opencl_clang.orig/CMakeLists.txt external_igc_opencl_clang/CMakeLists.txt
--- external_igc_opencl_clang.orig/CMakeLists.txt	2022-03-16 05:51:10 -0600
+++ external_igc_opencl_clang/CMakeLists.txt	2022-05-23 10:40:09 -0600
@@ -147,22 +147,24 @@
         )
     endif()
 
-
-    set(SPIRV_BASE_REVISION llvm_release_110)
-    set(TARGET_BRANCH "ocl-open-110")
-    get_filename_component(LLVM_MONOREPO_DIR ${LLVM_SOURCE_DIR} DIRECTORY)
-    set(LLVM_PATCHES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/patches/llvm
-                          ${CMAKE_CURRENT_SOURCE_DIR}/patches/clang)
-    apply_patches(${LLVM_MONOREPO_DIR}
-                  "${LLVM_PATCHES_DIRS}"
-                  ${LLVM_BASE_REVISION}
-                  ${TARGET_BRANCH}
-                  ret)
-    apply_patches(${SPIRV_SOURCE_DIR}
-                  ${CMAKE_CURRENT_SOURCE_DIR}/patches/spirv
-                  ${SPIRV_BASE_REVISION}
-                  ${TARGET_BRANCH}
-                  ret)
+    #
+    # Blender: Why apply these manually in igc.cmake
+    #
+    #set(SPIRV_BASE_REVISION llvm_release_110)
+    #set(TARGET_BRANCH "ocl-open-110")
+    #get_filename_component(LLVM_MONOREPO_DIR ${LLVM_SOURCE_DIR} DIRECTORY)
+    #set(LLVM_PATCHES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/patches/llvm
+    #                      ${CMAKE_CURRENT_SOURCE_DIR}/patches/clang)
+    #apply_patches(${LLVM_MONOREPO_DIR}
+    #              "${LLVM_PATCHES_DIRS}"
+    #              ${LLVM_BASE_REVISION}
+    #              ${TARGET_BRANCH}
+    #              ret)
+    #apply_patches(${SPIRV_SOURCE_DIR}
+    #              ${CMAKE_CURRENT_SOURCE_DIR}/patches/spirv
+    #              ${SPIRV_BASE_REVISION}
+    #              ${TARGET_BRANCH}
+    #              ret)
 endif(NOT USE_PREBUILT_LLVM)
 
 #
