diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e972e6..84f9674 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,21 +147,24 @@ if(NOT USE_PREBUILT_LLVM)
         )
     endif()
     
-    set(SPIRV_BASE_REVISION llvm_release_140)
-    set(TARGET_BRANCH "ocl-open-140")
-    get_filename_component(LLVM_MONOREPO_DIR ${LLVM_SOURCE_DIR} DIRECTORY)
-    set(LLVM_PATCHES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/patches/llvm
-                          ${CMAKE_CURRENT_SOURCE_DIR}/patches/clang)
-    apply_patches(${LLVM_MONOREPO_DIR}
-                  "${LLVM_PATCHES_DIRS}"
-                  ${LLVM_BASE_REVISION}
-                  ${TARGET_BRANCH}
-                  ret)
-    apply_patches(${SPIRV_SOURCE_DIR}
-                  ${CMAKE_CURRENT_SOURCE_DIR}/patches/spirv
-                  ${SPIRV_BASE_REVISION}
-                  ${TARGET_BRANCH}
-                  ret)
+    #
+    # Blender: We apply these manually in igc.cmake
+    #
+    #set(SPIRV_BASE_REVISION llvm_release_140)
+    #set(TARGET_BRANCH "ocl-open-140")
+    #get_filename_component(LLVM_MONOREPO_DIR ${LLVM_SOURCE_DIR} DIRECTORY)
+    #set(LLVM_PATCHES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/patches/llvm
+    #                      ${CMAKE_CURRENT_SOURCE_DIR}/patches/clang)
+    #apply_patches(${LLVM_MONOREPO_DIR}
+    #              "${LLVM_PATCHES_DIRS}"
+    #              ${LLVM_BASE_REVISION}
+    #              ${TARGET_BRANCH}
+    #              ret)
+    #apply_patches(${SPIRV_SOURCE_DIR}
+    #              ${CMAKE_CURRENT_SOURCE_DIR}/patches/spirv
+    #              ${SPIRV_BASE_REVISION}
+    #              ${TARGET_BRANCH}
+    #              ret)
 endif(NOT USE_PREBUILT_LLVM)
 
 #
