diff -Naur ispc-1.28.0/cmake/GenerateBuiltins.cmake external_ispc/cmake/GenerateBuiltins.cmake
--- ispc-1.28.0/cmake/GenerateBuiltins.cmake	2025-08-12 06:34:22.000000000 -0600
+++ external_ispc/cmake/GenerateBuiltins.cmake	2025-09-10 10:40:05.793808800 -0600
@@ -334,7 +334,7 @@
 
     add_custom_command(
         OUTPUT ${bc}
-        COMMAND cat ${input} | \"${LLVM_AS_EXECUTABLE}\" -o ${bc}
+        COMMAND ${CMAKE_COMMAND} -E cat ${input} | \"${LLVM_AS_EXECUTABLE}\" -o ${bc}
         DEPENDS ${input}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         )
