diff -Naur external_levelzero_org/CMakeLists.txt external_levelzero/CMakeLists.txt
--- external_levelzero_org/CMakeLists.txt       2022-03-07 13:22:11 -0700
+++ external_levelzero/CMakeLists.txt   2022-03-29 13:22:15 -0600
@@ -77,9 +77,6 @@
     #enabling Control Flow Guard
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /guard:cf")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /guard:cf")
-    # enable Spectre Mitigation
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Qspectre")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Qspectre")
 endif()

 #CXX compiler support