diff -Naur numpy-1.11.1/numpy/distutils/ccompiler.py numpy-1.11.1/numpy/distutils/ccompiler.py
--- numpy-1.11.1/numpy/distutils/ccompiler.py	2016-06-25 08:38:34 -0600
+++ numpy-1.11.1/numpy/distutils/ccompiler.py	2016-08-04 12:33:43 -0600
@@ -29,6 +29,7 @@
 
 # Using customized CCompiler.spawn.
 def CCompiler_spawn(self, cmd, display=None):
+    cmd = quote_args(cmd)
     """
     Execute a command in a sub-process.
 
diff -Naur numpy-1.11.1/numpy/distutils/misc_util.py numpy-1.11.1/numpy/distutils/misc_util.py
--- numpy-1.11.1/numpy/distutils/misc_util.py	2016-06-25 08:38:34 -0600
+++ numpy-1.11.1/numpy/distutils/misc_util.py	2016-08-04 12:34:56 -0600
@@ -116,7 +116,7 @@
     args = list(args)
     for i in range(len(args)):
         a = args[i]
-        if ' ' in a and a[0] not in '"\'':
+        if ' ' in a and a[0] not in '"\'' and a[len(a)-1] not in '"\'':
             args[i] = '"%s"' % (a)
     return args
 
