diff -Naur orig/devices/cuda/CMakeLists.txt external_openimagedenoise/devices/cuda/CMakeLists.txt
--- orig/devices/cuda/CMakeLists.txt
+++ external_openimagedenoise/devices/cuda/CMakeLists.txt
@@ -51,13 +52,15 @@
 set(OIDN_NVCC_SM75_FLAGS "-gencode arch=compute_75,code=sm_75")
 set(OIDN_NVCC_SM80_FLAGS "-gencode arch=compute_80,code=sm_80")
 set(OIDN_NVCC_SM90_FLAGS "-gencode arch=compute_90,code=sm_90")
+set(OIDN_NVCC_SM100_FLAGS "-gencode arch=compute_100,code=sm_100")
+set(OIDN_NVCC_SM120_FLAGS "-gencode arch=compute_120,code=sm_120")
 
 set_source_files_properties(
   cuda_conv.cu
   cuda_device.cu
   cuda_engine.cu
   PROPERTIES COMPILE_FLAGS
-    "${OIDN_NVCC_SM70_FLAGS} ${OIDN_NVCC_SM75_FLAGS} ${OIDN_NVCC_SM80_FLAGS} ${OIDN_NVCC_SM90_FLAGS}"
+    "${OIDN_NVCC_SM70_FLAGS} ${OIDN_NVCC_SM75_FLAGS} ${OIDN_NVCC_SM80_FLAGS} ${OIDN_NVCC_SM90_FLAGS} ${OIDN_NVCC_SM100_FLAGS} ${OIDN_NVCC_SM120_FLAGS}"
 )
 
 set_source_files_properties(
@@ -72,7 +75,7 @@
 
 set_source_files_properties(
   cutlass_conv_sm80.cu
-  PROPERTIES COMPILE_FLAGS "${OIDN_NVCC_SM80_FLAGS} ${OIDN_NVCC_SM90_FLAGS}"
+  PROPERTIES COMPILE_FLAGS "${OIDN_NVCC_SM80_FLAGS} ${OIDN_NVCC_SM90_FLAGS} ${OIDN_NVCC_SM100_FLAGS} ${OIDN_NVCC_SM120_FLAGS}"
 )
 
 add_library(OpenImageDenoise_device_cuda SHARED ${OIDN_CUDA_SOURCES} ${OIDN_GPU_SOURCES} ${OIDN_RESOURCE_FILE})
diff -Naur orig/devices/cuda/cuda_device.h external_openimagedenoise/devices/cuda/cuda_device.h
--- orig/devices/cuda/cuda_device.h
+++ external_openimagedenoise/devices/cuda/cuda_device.h
@@ -50,7 +50,7 @@
 
     // Supported compute capabilities
     static constexpr int minSMArch = 70;
-    static constexpr int maxSMArch = 99;
+    static constexpr int maxSMArch = 120;
 
     int deviceID = 0;
   #if defined(OIDN_DEVICE_CUDA_API_DRIVER)
