--- external_openimagedenoise/cmake/oidn_ispc.cmake	2021-02-15 17:29:34.000000000 +0100
+++ external_openimagedenoise/cmake/oidn_ispc.cmake2	2021-02-15 17:29:28.000000000 +0100
@@ -98,7 +98,7 @@
   elseif(OIDN_ARCH STREQUAL "ARM64")
     set(ISPC_ARCHITECTURE "aarch64")
     if(APPLE)
-      set(ISPC_TARGET_OS "--target-os=ios")
+      set(ISPC_TARGET_OS "--target-os=macos")
     endif()
   endif()
