diff --git a/src/include/OpenImageIO/simd.h b/src/include/OpenImageIO/simd.h
index b1701946f..d148a89d3 100644
--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -119,6 +119,9 @@
     // Cuda -- don't include any of these headers
 #elif defined(_WIN32)
 #  include <intrin.h>
+#  if defined(__ARM_NEON__) && !defined(OIIO_NO_NEON)
+#    include <arm_neon.h>
+#  endif
 #elif defined(__GNUC__) && (defined(__x86_64__) || defined(__i386__)) || defined(__e2k__)
 #  include <x86intrin.h>
 #elif defined(__GNUC__) && defined(__ARM_NEON__) && !defined(OIIO_NO_NEON)
