diff -Naur openvdb-12.0.0/openvdb/openvdb/CMakeLists.txt openvdb/openvdb/openvdb/CMakeLists.txt
--- openvdb-12.0.0/openvdb/openvdb/CMakeLists.txt	2024-10-31 16:59:05.000000000 -0700
+++ openvdb/openvdb/openvdb/CMakeLists.txt	2024-12-11 08:47:05.635327200 -0800
@@ -148,6 +148,15 @@
   endif()
 endif()
 
+add_compile_definitions(
+  #Blender: prevent implicit TBB linkage 
+  __TBB_NO_IMPLICIT_LINKAGE 
+  #Blender: OpenEXR is static 
+  OPENVDB_OPENEXR_STATICLIB 
+  #Blender: Link the debug python libs rather than the release ones
+  $<$<CONFIG:Debug>:Py_DEBUG>
+)
+
 if(USE_BLOSC)
   # Find Blosc libraries
   find_package(Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED)
@@ -608,7 +617,12 @@
 # imported targets.
 
 if(OPENVDB_CORE_SHARED)
-  add_library(openvdb_shared SHARED ${OPENVDB_LIBRARY_SOURCE_FILES})
+  if(WIN32)
+    configure_file(version.rc.in ${CMAKE_CURRENT_BINARY_DIR}/version.rc @ONLY)
+    add_library(openvdb_shared SHARED ${OPENVDB_LIBRARY_SOURCE_FILES} ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
+  else()
+    add_library(openvdb_shared SHARED ${OPENVDB_LIBRARY_SOURCE_FILES})
+  endif()
 endif()
 
 if(OPENVDB_CORE_STATIC)
diff -Naur openvdb-12.0.0/openvdb/openvdb/python/CMakeLists.txt openvdb/openvdb/openvdb/python/CMakeLists.txt
--- openvdb-12.0.0/openvdb/openvdb/python/CMakeLists.txt	2024-10-31 16:59:05.000000000 -0700
+++ openvdb/openvdb/openvdb/python/CMakeLists.txt	2024-12-11 09:02:05.377733900 -0800
@@ -22,7 +22,9 @@
 option(OPENVDB_BUILD_PYTHON_UNITTESTS [=[
 "Include the OpenVDB Python unit test. Requires a python interpreter]=]
 ${OPENVDB_BUILD_UNITTESTS})
-
+#blender: prevent double _d for the python module 
+set(CMAKE_DEBUG_POSTFIX "")
+add_compile_definitions($<$<CONFIG:Debug>:Py_DEBUG> __TBB_NO_IMPLICIT_LINKAGE)
 #########################################################################
 
 message(STATUS "----------------------------------------------------")
diff -Naur openvdb-12.0.0/openvdb/openvdb/version.rc.in openvdb/openvdb/openvdb/version.rc.in
--- openvdb-12.0.0/openvdb/openvdb/version.rc.in	1969-12-31 16:00:00.000000000 -0800
+++ openvdb/openvdb/openvdb/version.rc.in	2024-12-11 08:47:05.697536200 -0800
@@ -0,0 +1,48 @@
+#include <winver.h>
+
+#define VER_FILEVERSION             @OpenVDB_MAJOR_VERSION@,@OpenVDB_MINOR_VERSION@,@OpenVDB_PATCH_VERSION@,0
+#define VER_FILEVERSION_STR         "@OpenVDB_MAJOR_VERSION@.@OpenVDB_MINOR_VERSION@.@OpenVDB_PATCH_VERSION@.0\0"
+
+#define VER_PRODUCTVERSION          @OpenVDB_MAJOR_VERSION@,@OpenVDB_MINOR_VERSION@,@OpenVDB_PATCH_VERSION@,0
+#define VER_PRODUCTVERSION_STR      "@OpenVDB_MAJOR_VERSION@.@OpenVDB_MINOR_VERSION@\0"
+
+#ifndef DEBUG
+#define VER_DEBUG                   0
+#else
+#define VER_DEBUG                   VS_FF_DEBUG
+#endif
+
+VS_VERSION_INFO VERSIONINFO
+FILEVERSION     VER_FILEVERSION
+PRODUCTVERSION  VER_PRODUCTVERSION
+FILEFLAGSMASK   VS_FFI_FILEFLAGSMASK
+FILEFLAGS       (VER_DEBUG)
+FILEOS          VOS__WINDOWS32
+FILETYPE        VFT_DLL
+FILESUBTYPE     VFT2_UNKNOWN
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904E4"
+        BEGIN
+            VALUE "FileDescription",  "OpenVDB"
+            VALUE "FileVersion",      VER_FILEVERSION_STR
+            VALUE "InternalName",     "OpenVDB"
+            VALUE "ProductName",      "OpenVDB"
+            VALUE "ProductVersion",   VER_PRODUCTVERSION_STR
+        END
+    END
+
+    BLOCK "VarFileInfo"
+    BEGIN
+        /* The following line should only be modified for localized versions.     */
+        /* It consists of any number of WORD,WORD pairs, with each pair           */
+        /* describing a language,codepage combination supported by the file.      */
+        /*                                                                        */
+        /* For example, a file might have values "0x409,1252" indicating that it  */
+        /* supports English language (0x409) in the Windows ANSI codepage (1252). */
+
+        VALUE "Translation", 0x409, 1252
+
+    END
+END
