diff -Naur external_openimageio.orig/src/cmake/compiler.cmake external_openimageio/src/cmake/compiler.cmake
--- external_openimageio.orig/src/cmake/compiler.cmake	2018-07-31 23:45:19 -0600
+++ external_openimageio/src/cmake/compiler.cmake	2018-08-16 12:50:12 -0600
@@ -152,6 +152,7 @@
     add_definitions (-D_CRT_NONSTDC_NO_WARNINGS)
     add_definitions (-D_SCL_SECURE_NO_WARNINGS)
     add_definitions (-DJAS_WIN_MSVC_BUILD)
+    add_definitions (-DOPJ_STATIC)
 endif (MSVC)
 
 # Use ccache if found
