Index: OpenImageIO/osdep.h
===================================================================
--- OpenImageIO/osdep.h	(revision 61595)
+++ OpenImageIO/osdep.h	(working copy)
@@ -34,6 +34,7 @@
 # define WIN32_LEAN_AND_MEAN
 # define VC_EXTRALEAN
 # define NOMINMAX
+# define NOGDI
 # include <windows.h>
 #endif
 
Index: OpenImageIO/platform.h
===================================================================
--- OpenImageIO/platform.h	(revision 61595)
+++ OpenImageIO/platform.h	(working copy)
@@ -77,6 +77,9 @@
 # ifndef NOMINMAX
 #   define NOMINMAX
 # endif
+# ifndef NOGDI
+#   define NOGDI
+# endif
 # include <windows.h>
 #endif
 
