--- CMakeLists.txt	2016-11-01 01:03:44 -0600
+++ CMakeLists.txt	2016-12-01 09:20:12 -0700
@@ -454,7 +454,7 @@
     add_definitions (-D_CRT_NONSTDC_NO_WARNINGS)
     add_definitions (-D_SCL_SECURE_NO_WARNINGS)
     add_definitions (-DJAS_WIN_MSVC_BUILD)
-    add_definitions (-DOPENEXR_DLL)
+    add_definitions (-DOPJ_STATIC)
     if (LINKSTATIC)
         add_definitions (-DBoost_USE_STATIC_LIBS=1)
     else ()
