--- openvdb/tree/LeafNode.h	2015-10-01 15:55:33 -0600
+++ openvdb/tree/LeafNode.h	2016-03-26 13:12:22 -0600
@@ -70,13 +70,14 @@
     typedef boost::shared_ptr<LeafNode>  Ptr;
     typedef util::NodeMask<Log2Dim>      NodeMaskType;
 
-    static const Index
-        LOG2DIM     = Log2Dim,      // needed by parent nodes
-        TOTAL       = Log2Dim,      // needed by parent nodes
-        DIM         = 1 << TOTAL,   // dimension along one coordinate direction
-        NUM_VALUES  = 1 << 3 * Log2Dim,
-        NUM_VOXELS  = NUM_VALUES,   // total number of voxels represented by this node
-        SIZE        = NUM_VALUES,
+	static const Index
+		LOG2DIM = Log2Dim,      // needed by parent nodes
+		TOTAL = Log2Dim,      // needed by parent nodes
+		DIM = 1 << TOTAL,   // dimension along one coordinate direction
+		NUM_VALUES = 1 << 3 * Log2Dim,
+		NUM_VOXELS = NUM_VALUES;   // total number of voxels represented by this node
+	static const Index
+		SIZE        = NUM_VALUES,
         LEVEL       = 0;            // level 0 = leaf
 
     /// @brief ValueConverter<T>::Type is the type of a LeafNode having the same
--- openvdb/PlatformConfig.h	2016-03-30 15:09:49 -0600
+++ openvdb/PlatformConfig.h	2016-04-01 07:00:38 -0600
@@ -47,7 +47,7 @@
     #if !defined(OPENVDB_OPENEXR_STATICLIB) && !defined(OPENEXR_DLL)
         #define OPENEXR_DLL
     #endif
-
+    #define NOMINMAX
 #endif // _WIN32
 
 #endif // OPENVDB_PLATFORMCONFIG_HAS_BEEN_INCLUDED
