diff -Naur OpenShadingLanguage-Release-1.9.9/src/cmake/flexbison.cmake.rej external_osl/src/cmake/flexbison.cmake.rej
--- OpenShadingLanguage-Release-1.9.9/src/cmake/flexbison.cmake.rej	1969-12-31 17:00:00 -0700
+++ external_osl/src/cmake/flexbison.cmake.rej	2018-08-24 17:42:11 -0600
@@ -0,0 +1,11 @@
+--- src/cmake/flexbison.cmake	2018-05-01 16:39:02 -0600
++++ src/cmake/flexbison.cmake	2018-08-24 10:24:03 -0600
+@@ -77,7 +77,7 @@
+           DEPENDS ${${compiler_headers}}
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+         ADD_CUSTOM_COMMAND ( OUTPUT ${flexoutputcxx} 
+-          COMMAND ${FLEX_EXECUTABLE} -o ${flexoutputcxx} "${CMAKE_CURRENT_SOURCE_DIR}/${flexsrc}"
++          COMMAND ${FLEX_EXECUTABLE} ${FLEX_EXTRA_OPTIONS} -o ${flexoutputcxx} "${CMAKE_CURRENT_SOURCE_DIR}/${flexsrc}"
+           MAIN_DEPENDENCY ${flexsrc}
+           DEPENDS ${${compiler_headers}}
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
diff -Naur OpenShadingLanguage-Release-1.9.9/src/include/OSL/llvm_util.h external_osl/src/include/OSL/llvm_util.h
--- OpenShadingLanguage-Release-1.9.9/src/include/OSL/llvm_util.h	2018-05-01 16:39:02 -0600
+++ external_osl/src/include/OSL/llvm_util.h	2018-08-25 14:05:00 -0600
@@ -33,6 +33,8 @@
 
 #include <vector>
 
+#define OSL_HAS_BLENDER_CLEANUP_FIX
+
 #ifdef LLVM_NAMESPACE
 namespace llvm = LLVM_NAMESPACE;
 #endif
@@ -487,6 +489,7 @@
     std::string func_name (llvm::Function *f);
 
     static size_t total_jit_memory_held ();
+    static void Cleanup ();
 
 private:
     class MemoryManager;
diff -Naur OpenShadingLanguage-Release-1.9.9/src/liboslexec/llvm_util.cpp external_osl/src/liboslexec/llvm_util.cpp
--- OpenShadingLanguage-Release-1.9.9/src/liboslexec/llvm_util.cpp	2018-05-01 16:39:02 -0600
+++ external_osl/src/liboslexec/llvm_util.cpp	2018-08-25 14:04:27 -0600
@@ -140,7 +140,10 @@
 };
 
 
-
+void LLVM_Util::Cleanup ()
+{
+       jitmm_hold.clear();
+}
 
 size_t
 LLVM_Util::total_jit_memory_held ()
diff -Naur OpenShadingLanguage-Release-1.9.9/CMakeLists.txt external_osl/CMakeLists.txt
 --- orig/CMakeLists.txt	2020-01-27 16:22:31 -0700
+++ external_osl/CMakeLists.txt	2020-05-13 18:04:52 -0600
@@ -102,10 +102,11 @@
 set (OPTIX_EXTRA_LIBS CACHE STRING "Extra lib targets needed for OptiX")
 set (CUDA_TARGET_ARCH "sm_35" CACHE STRING "CUDA GPU architecture (e.g. sm_35)")
 
-# set (USE_OIIO_STATIC ON CACHE BOOL "If OIIO is built static")
-# if (USE_OIIO_STATIC)
-#     add_definitions ("-DOIIO_STATIC_BUILD=1")
-# endif ()
+set (USE_OIIO_STATIC ON CACHE BOOL "If OIIO is built static")
+if (USE_OIIO_STATIC)
+    add_definitions ("-DOIIO_STATIC_BUILD=1")
+    add_definitions ("-DOIIO_STATIC_DEFINE=1")
+endif ()
 
 set (OSL_NO_DEFAULT_TEXTURESYSTEM OFF CACHE BOOL "Do not use create a raw OIIO::TextureSystem")
 if (OSL_NO_DEFAULT_TEXTURESYSTEM)