--- CMakeLists.txt	2016-10-31 16:48:19 -0600
+++ CMakeLists.txt	2017-04-10 10:38:48 -0600
@@ -269,6 +269,11 @@
     add_definitions ("-DOIIO_STATIC_BUILD=1")
 endif ()
 
+set (OIIO_NOSIMD OFF CACHE BOOL "Disable simd support in oiio")
+if (OIIO_NOSIMD)
+    add_definitions ("-DOIIO_NO_SSE=1")
+endif ()
+
 if (OSL_NO_DEFAULT_TEXTURESYSTEM)
     add_definitions ("-DOSL_NO_DEFAULT_TEXTURESYSTEM=1")
 endif ()
