diff --git a/Lib/ctypes/__init__.py b/Lib/ctypes/__init__.py
index 7fc1181f25a..8be3b2954ce 100644
--- a/Lib/ctypes/__init__.py
+++ b/Lib/ctypes/__init__.py
@@ -488,9 +488,6 @@ def LoadLibrary(self, name):
 
 if _os.name == "nt":
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
-elif _sys.platform in ["android", "cygwin"]:
-    # These are Unix-like platforms which use a dynamically-linked libpython.
-    pythonapi = PyDLL(_sysconfig.get_config_var("LDLIBRARY"))
 else:
     pythonapi = PyDLL(None)
 
diff --git a/Lib/platform.py b/Lib/platform.py
index 8895177e326..619be66d352 100755
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -577,13 +577,13 @@ def android_ver(release="", api_level=0, manufacturer="", model="", device="",
     if sys.platform == "android":
         try:
             from ctypes import CDLL, c_char_p, create_string_buffer
-        except ImportError:
+            system_property_get = getattr(CDLL("libc.so"), "__system_property_get")
+        except (ImportError, OSError):
             pass
         else:
             # An NDK developer confirmed that this is an officially-supported
             # API (https://stackoverflow.com/a/28416743). Use `getattr` to avoid
             # private name mangling.
-            system_property_get = getattr(CDLL("libc.so"), "__system_property_get")
             system_property_get.argtypes = (c_char_p, c_char_p)
 
             def getprop(name, default):
