--- pyconfig.h.in	2017-09-19 14:41:01.000000000 +0300
+++ pyconfig.h.in	2017-09-19 14:43:18.000000000 +0300
@@ -119,12 +119,6 @@
 /* Define to 1 if you have the `clock' function. */
 #undef HAVE_CLOCK
 
-/* Define to 1 if you have the `clock_getres' function. */
-#undef HAVE_CLOCK_GETRES
-
-/* Define to 1 if you have the `clock_gettime' function. */
-#undef HAVE_CLOCK_GETTIME
-
 /* Define to 1 if you have the `clock_settime' function. */
 #undef HAVE_CLOCK_SETTIME
 
@@ -369,9 +363,6 @@
 /* Define this if you have flockfile(), getc_unlocked(), and funlockfile() */
 #undef HAVE_GETC_UNLOCKED
 
-/* Define to 1 if you have the `getentropy' function. */
-#undef HAVE_GETENTROPY
-
 /* Define to 1 if you have the `getgrouplist' function. */
 #undef HAVE_GETGROUPLIST
 
