--- Include/Python.h	2017-07-07 21:33:27 -0600
+++ Include/Python.h	2017-09-19 10:36:10 -0600
@@ -2,6 +2,10 @@
 #define Py_PYTHON_H
 /* Since this is a "meta-include" file, no #ifdef __cplusplus / extern "C" { */
 
+#if _MSC_VER < 1900
+#define inline __inline
+#endif
+
 /* Include nearly all Python header files */
 
 #include "patchlevel.h"
--- Include/pydtrace.h	2017-07-07 21:33:27 -0600
+++ Include/pydtrace.h	2017-09-19 10:32:31 -0600
@@ -2,6 +2,11 @@
 
 #ifndef Py_DTRACE_H
 #define Py_DTRACE_H
+
+#if _MSC_VER < 1900
+#define inline __inline
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif

--- Modules/_blake2/impl/blake2.h	2017-07-07 21:33:27 -0600
+++ Modules/_blake2/impl/blake2.h	2017-09-19 10:22:41 -0600
@@ -19,6 +19,10 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if _MSC_VER < 1900
+#define inline __inline 
+#endif
+
 #ifdef BLAKE2_NO_INLINE
 #define BLAKE2_LOCAL_INLINE(type) static type
 #endif

--- pcbuild/build.bat   2016-05-21 09:53:55 -0600
+++ pcbuild/build.bat   2016-05-21 09:56:16 -0600
@@ -59,6 +59,7 @@
 if "%~1"=="-h" goto Usage
 if "%~1"=="-c" (set conf=%2) & shift & shift & goto CheckOpts
 if "%~1"=="-p" (set platf=%2) & shift & shift & goto CheckOpts
+if "%~1"=="-k" (set vs_toolset=%2) & shift & shift & goto CheckOpts
 if "%~1"=="-r" (set target=Rebuild) & shift & goto CheckOpts
 if "%~1"=="-t" (set target=%2) & shift & shift & goto CheckOpts
 if "%~1"=="-d" (set conf=Debug) & shift & goto CheckOpts
@@ -120,7 +120,7 @@

 :Kill
 echo on
-%MSBUILD% "%dir%\pythoncore.vcxproj" /t:KillPython %verbose%^
+%MSBUILD% "%dir%\pythoncore.vcxproj" /t:KillPython %verbose% /p:PlatformToolset=%vs_toolset%^
  /p:Configuration=%conf% /p:Platform=%platf%^
  /p:KillPython=true

@@ -130,7 +130,7 @@
 rem batch is, shall we say, "lackluster"
 echo on
 %MSBUILD% "%dir%pcbuild.proj" /t:%target% %parallel% %verbose%^
- /p:Configuration=%conf% /p:Platform=%platf%^
+ /p:Configuration=%conf% /p:Platform=%platf% /p:PlatformToolset=%vs_toolset%^
  /p:IncludeExternals=%IncludeExternals%^
  /p:IncludeSSL=%IncludeSSL% /p:IncludeTkinter=%IncludeTkinter%^
  /p:UseTestMarker=%UseTestMarker%^

--- pcbuild/sqlite3.vcxproj	2015-12-06 18:39:10 -0700
+++ pcbuild/sqlite3.vcxproj	2016-11-02 09:25:56 -0600
@@ -43,7 +43,7 @@
   <Import Project="python.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
-    <ConfigurationType>DynamicLibrary</ConfigurationType>
+    <ConfigurationType>StaticLibrary</ConfigurationType>
     <CharacterSet>NotSet</CharacterSet>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
