--- _msvccompiler.py.orig	2017-05-20 19:31:45 -0600
+++ _msvccompiler.py	2017-06-10 10:05:38 -0600
@@ -222,9 +222,9 @@
         # use /MT[d] to build statically, then switch from libucrt[d].lib to ucrt[d].lib
         # later to dynamically link to ucrtbase but not vcruntime.
         self.compile_options = [
-            '/nologo', '/Ox', '/W3', '/GL', '/DNDEBUG'
+            '/nologo', '/Ox', '/W3', '/GL', '/DNDEBUG' , '/MD'
         ]
-        self.compile_options.append('/MD' if self._vcruntime_redist else '/MT')
+        #self.compile_options.append('/MD' if self._vcruntime_redist else '/MT')
 
         self.compile_options_debug = [
             '/nologo', '/Od', '/MDd', '/Zi', '/W3', '/D_DEBUG'
@@ -233,11 +233,11 @@
         ldflags = [
             '/nologo', '/INCREMENTAL:NO', '/LTCG'
         ]
-        if not self._vcruntime_redist:
-            ldflags.extend(('/nodefaultlib:libucrt.lib', 'ucrt.lib'))
+        #if not self._vcruntime_redist:
+        #    ldflags.extend(('/nodefaultlib:libucrt.lib', 'ucrt.lib'))
 
         ldflags_debug = [
-            '/nologo', '/INCREMENTAL:NO', '/LTCG', '/DEBUG:FULL'
+            '/nologo', '/INCREMENTAL:NO', '/LTCG'
         ]
 
         self.ldflags_exe = [*ldflags, '/MANIFEST:EMBED,ID=1']
