--- configure.orig	2012-01-22 19:06:43 -0700
+++ configure	2016-04-06 20:00:50 -0600
@@ -16492,10 +16492,10 @@
 	HAVE_ORC=yes
 fi
 if test "x${HAVE_ORC}" != xyes ; then
-  as_fn_error $? "orc-0.4 >= $ORC_VER is required" "$LINENO" 5
+  $as_echo "orc-0.4 >= $ORC_VER is required" 
 fi
 SCHRO_PKG_DEPS="$SCHRO_PKG_DEPS orc-0.4 >= $ORC_VER"
-ORCC=`$PKG_CONFIG --variable=orcc orc-0.4`
+#ORCC=`$PKG_CONFIG --variable=orcc orc-0.4`
 
  if test "x$cross_compiling" != xyes; then
   HAVE_ORCC_TRUE=
--- Makefile.in	2012-01-22 18:06:42 -0700
+++ Makefile.in	2016-04-06 20:30:09 -0600
@@ -291,7 +291,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = schroedinger doc tools testsuite
+SUBDIRS = schroedinger doc tools 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 DIST_SUBDIRS = schroedinger doc tools testsuite
 EXTRA_DIST = COPYING COPYING.GPL COPYING.LGPL COPYING.MIT COPYING.MPL \

--- schroedinger.pc.in  2011-03-21 17:08:39 -0600
+++ schroedinger.pc.in  2016-04-08 13:30:42 -0600
@@ -7,9 +7,9 @@

 Name: schroedinger-@SCHRO_MAJORMINOR@
 Description: Dirac codec library
-Requires.private: @SCHRO_PKG_DEPS@
+Requires: @SCHRO_PKG_DEPS@
 Version: @VERSION@
-Libs: -L${libdir} -lschroedinger-@SCHRO_MAJORMINOR@
+Libs: -L${libdir} -lschroedinger-@SCHRO_MAJORMINOR@ -lorc-0.4
 Libs.private: @PTHREAD_LIBS@ @LIBM@
 Cflags: -I${includedir}
 
--- ./schroedinger/schrodecoder.c	2012-01-23 00:38:57.000000000 +0100
+++ ./schroedinger/schrodecoder.c	2016-05-15 06:07:24.000000000 +0200
@@ -70,8 +70,8 @@
 };
 
 
-int _schro_decode_prediction_only;
-int _schro_telemetry;
+int _schro_decode_prediction_only = 0;
+int _schro_telemetry = 0;
 
 static void schro_decoder_x_decode_motion (SchroAsyncStage * stage);
 static void schro_decoder_x_render_motion (SchroAsyncStage * stage);
