diff --git a/pxr/imaging/hgiVulkan/computeCmds.cpp b/pxr/imaging/hgiVulkan/computeCmds.cpp
index cbf9704fd..74a81c787 100644
--- a/pxr/imaging/hgiVulkan/computeCmds.cpp
+++ b/pxr/imaging/hgiVulkan/computeCmds.cpp
@@ -119,10 +119,11 @@ HgiVulkanComputeCmds::Dispatch(int dimX, int dimY)
     // Determine device's num compute work group limits
     const VkPhysicalDeviceLimits limits = 
         _hgi->GetCapabilities()->vkDeviceProperties.limits;
+    const uint32_t maxAllowedLimit = (1 << 31) - 1;
     const GfVec3i maxNumWorkGroups = GfVec3i(
-        limits.maxComputeWorkGroupCount[0],
-        limits.maxComputeWorkGroupCount[1],
-        limits.maxComputeWorkGroupCount[2]);
+        std::min<uint32_t>(maxAllowedLimit, limits.maxComputeWorkGroupCount[0]),
+        std::min<uint32_t>(maxAllowedLimit, limits.maxComputeWorkGroupCount[1]),
+        std::min<uint32_t>(maxAllowedLimit, limits.maxComputeWorkGroupCount[2]));
 
     if (numWorkGroupsX > maxNumWorkGroups[0]) {
         TF_WARN("Max number of work group available from device is %i, larger "
