diff --git a/pxr/base/arch/daemon.cpp b/pxr/base/arch/daemon.cpp
index ef1403493..44ace356b 100644
--- a/pxr/base/arch/daemon.cpp
+++ b/pxr/base/arch/daemon.cpp
@@ -40,7 +40,14 @@ ArchCloseAllFiles(int nExcept, const int* exceptFds)
 
     if (limits.rlim_cur == RLIM_INFINITY)
     {
+#if defined(__ANDROID__)
+        // Unlike other libcs, Android Bionic does not define NOFILE in
+        // sys/params.h. Fallback requesting _SC_OPEN_MAX via syconf,
+        // similarly to what Bionic does internally in libc/bionic/ndk_cruft.cpp:~290
+        maxfd = sysconf(_SC_OPEN_MAX);
+#else
         maxfd = NOFILE;
+#endif
     }
     else
     {
diff --git a/pxr/base/arch/stackTrace.cpp b/pxr/base/arch/stackTrace.cpp
index c196392fd..7f8d6f6a3 100644
--- a/pxr/base/arch/stackTrace.cpp
+++ b/pxr/base/arch/stackTrace.cpp
@@ -704,7 +704,9 @@ static int
 nonLockingExecv(const char *path, char *const argv[])
 {
 #if defined(ARCH_OS_LINUX)
-     return nonLockingLinux__execve (path, argv, __environ);
+     /* Android Bionic libc only exposes `environ`, not `__environ`,
+        which is an alias to `environ` in glibc. */
+     return nonLockingLinux__execve (path, argv, environ);
 #else
      return execv(path, argv);
 #endif
