diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index c74db2179..bb8fb3c29 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -378,7 +378,7 @@ if(GCC)
             endif()
         endif()
     endif(ENABLE_AGGRESSIVE_CHECKS)
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE CC_VERSION)
+    #execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE CC_VERSION)
 endif(GCC)
 
 find_package(Nasm)
diff --git a/source/dynamicHDR10/CMakeLists.txt b/source/dynamicHDR10/CMakeLists.txt
index 22fb79d44..ad0f3a7a8 100644
--- a/source/dynamicHDR10/CMakeLists.txt
+++ b/source/dynamicHDR10/CMakeLists.txt
@@ -140,7 +140,7 @@ if(GCC)
             endif()
         endif()
     endif(ENABLE_AGGRESSIVE_CHECKS)
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE CC_VERSION)
+    #execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE CC_VERSION)
 endif(GCC)
 
 # Build options
diff --git a/source/common/cpu.cpp b/source/common/cpu.cpp
index ae0907890..4d4c7dd97 100644
--- a/source/common/cpu.cpp
+++ b/source/common/cpu.cpp
@@ -61,6 +61,10 @@ static void sigill_handler(int sig)

 #endif // if X265_ARCH_ARM

+#if X265_ARCH_ARM64
+#include "aarch64/cpu.h"
+#endif // if X265_ARCH_ARM64
+
 namespace X265_NS {
 static bool enable512 = false;
 const cpu_name_t cpu_names[] =
@@ -390,7 +394,6 @@ uint32_t cpu_detect(bool benableavx512)
 }

 #elif X265_ARCH_ARM64
-#include "aarch64/cpu.h"

 uint32_t cpu_detect(bool benableavx512)
 {
