diff --git a/zconf.h.cmakein b/zconf.h.cmakein
index 247ba24..e5c16e6 100644
--- a/zconf.h.cmakein
+++ b/zconf.h.cmakein
@@ -439,7 +439,7 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_UNISTD_H
 #endif
 
