diff --git a/internal/ceres/collections_port.h b/internal/ceres/collections_port.h
index a356cc0..c2fce90 100644
--- a/internal/ceres/collections_port.h
+++ b/internal/ceres/collections_port.h
@@ -77,7 +77,7 @@ struct HashMap : std::tr1::unordered_map<K, V> {};
 template<typename K>
 struct HashSet : std::tr1::unordered_set<K> {};
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW64__) && !defined(__MINGW32__)
 #define GG_LONGLONG(x) x##I64
 #define GG_ULONGLONG(x) x##UI64
 #else
