Index: extern/libopenjpeg/opj_malloc.h
===================================================================
--- extern/libopenjpeg/opj_malloc.h	(revision 27736)
+++ extern/libopenjpeg/opj_malloc.h	(working copy)
@@ -76,7 +76,7 @@
 	#if defined(__sun)
 			#define HAVE_MEMALIGN
 		#elif defined(__GNUC__)
-                        #ifndef __APPLE__
+                        #if !defined(__APPLE__) && !defined(__FreeBSD__)
 			    #define HAVE_MEMALIGN
 			    #include <malloc.h>
                         #endif
