Index: opj_malloc.h
===================================================================
--- opj_malloc.h	(revision 15089)
+++ opj_malloc.h	(working copy)
@@ -76,8 +76,10 @@
 	#if defined(__sun)
 			#define HAVE_MEMALIGN
 		#elif defined(__GNUC__)
-			#define HAVE_MEMALIGN
-			#include <malloc.h>		
+                        #ifndef __APPLE__
+			    #define HAVE_MEMALIGN
+			    #include <malloc.h>
+                        #endif
 		/* Linux x86_64 and OSX always align allocations to 16 bytes */
 		#elif !defined(__amd64__) && !defined(__APPLE__)	
 			/* FIXME: Yes, this is a big assumption */
