diff -r 47dfdaff1dd5 include/carve/csg_triangulator.hpp
--- a/include/carve/csg_triangulator.hpp	Thu Jan 12 15:49:04 2012 -0500
+++ b/include/carve/csg_triangulator.hpp	Fri Jan 13 03:13:32 2012 +0600
@@ -174,6 +174,7 @@
 
       double scoreQuad(edge_map_t::iterator i, edge_map_t &edge_map) {
         if (!(*i).second.first || !(*i).second.second) return -1;
+        return 0;
       }
 
       carve::mesh::MeshSet<3>::face_t *mergeQuad(edge_map_t::iterator i, edge_map_t &edge_map) {
diff -r 47dfdaff1dd5 include/carve/exact.hpp
--- a/include/carve/exact.hpp	Thu Jan 12 15:49:04 2012 -0500
+++ b/include/carve/exact.hpp	Fri Jan 13 03:13:32 2012 +0600
@@ -379,7 +379,7 @@
         prod_2_1(b, a, r);
       }
 
-      static inline double prod_4_1(const double *a, const double *b, double *r) {
+      static inline void prod_4_1(const double *a, const double *b, double *r) {
         double b_sp[2]; split(b[0], b_sp);
         double t1[2]; prod_1_1s(a+0, b, b_sp, t1);
         r[0] = t1[0];
@@ -639,8 +639,9 @@
     }
 
 
-    exact_t operator+(const exact_t &a, const exact_t &b) {
-    }
+    // XXX: not implemented yet
+    //exact_t operator+(const exact_t &a, const exact_t &b) {
+    //}
 
 
 
diff -r 47dfdaff1dd5 src/selfintersect.cpp
--- a/src/selfintersect.cpp	Thu Jan 12 15:49:04 2012 -0500
+++ b/src/selfintersect.cpp	Fri Jan 13 03:13:32 2012 +0600
@@ -465,6 +465,7 @@
 
 // returns true if no intersection, based upon edge^a_i and edge^b_j separating axis.
 bool sat_edge(const vec3 tri_a[3], const vec3 tri_b[3], unsigned i, unsigned j) {
+  return false;
 }
 
 
