diff --git a/extern/ceres/internal/ceres/sparse_cholesky.cc b/extern/ceres/internal/ceres/sparse_cholesky.cc
index 0639ea90664..d9d2100d3f9 100644
--- a/internal/ceres/sparse_cholesky.cc
+++ b/internal/ceres/sparse_cholesky.cc
@@ -56,6 +56,7 @@ std::unique_ptr<SparseCholesky> SparseCholesky::Create(
       }
       break;
 #else
+      (void)ordering_type;
       LOG(FATAL) << "Ceres was compiled without support for SuiteSparse.";
 #endif
 
