diff --git a/src/libmv/logging/logging.h b/src/libmv/logging/logging.h
index 067da52..af86c4b 100644
--- a/src/libmv/logging/logging.h
+++ b/src/libmv/logging/logging.h
@@ -21,7 +21,7 @@
 #ifndef LIBMV_LOGGING_LOGGING_H
 #define LIBMV_LOGGING_LOGGING_H
 
-#include "third_party/glog/src/glog/logging.h"
+#include "glog/logging.h"
 
 #define LG LOG(INFO)
 #define V0 LOG(INFO)
diff --git a/src/libmv/simple_pipeline/pipeline.cc b/src/libmv/simple_pipeline/pipeline.cc
index 2459d05..2e4e5a6 100644
--- a/src/libmv/simple_pipeline/pipeline.cc
+++ b/src/libmv/simple_pipeline/pipeline.cc
@@ -280,25 +280,6 @@ double InternalReprojectionError(const Tracks &image_tracks,
     double ex = reprojected_marker.x - markers[i].x;
     double ey = reprojected_marker.y - markers[i].y;
 
-    const int N = 100;
-    char line[N];
-    snprintf(line, N,
-           "image %-3d track %-3d "
-           "x %7.1f y %7.1f "
-           "rx %7.1f ry %7.1f "
-           "ex %7.1f ey %7.1f"
-           "    e %7.1f",
-           markers[i].image,
-           markers[i].track,
-           markers[i].x,
-           markers[i].y,
-           reprojected_marker.x,
-           reprojected_marker.y,
-           ex,
-           ey,
-           sqrt(ex*ex + ey*ey));
-    LG << line;
-
     total_error += sqrt(ex*ex + ey*ey);
   }
   LG << "Skipped " << num_skipped << " markers.";
diff --git a/src/third_party/glog/src/glog/logging.h b/src/third_party/glog/src/glog/logging.h
index 57615ef..a58d478 100644
--- a/src/third_party/glog/src/glog/logging.h
+++ b/src/third_party/glog/src/glog/logging.h
@@ -33,6 +33,7 @@
 // Pretty much everybody needs to #include this file so that they can
 // log various happenings.
 //
+
 #ifndef _LOGGING_H_
 #define _LOGGING_H_
 
diff --git a/src/third_party/glog/src/logging.cc b/src/third_party/glog/src/logging.cc
index 868898f..1bb3867 100644
--- a/src/third_party/glog/src/logging.cc
+++ b/src/third_party/glog/src/logging.cc
@@ -58,8 +58,8 @@
 #include <errno.h>                   // for errno
 #include <sstream>
 #include "base/commandlineflags.h"        // to get the program name
-#include "glog/logging.h"
-#include "glog/raw_logging.h"
+#include <glog/logging.h>
+#include <glog/raw_logging.h>
 #include "base/googleinit.h"
 
 #ifdef HAVE_STACKTRACE
@@ -1232,7 +1232,9 @@ void LogMessage::RecordCrashReason(
 }
 
 static void logging_fail() {
-#if defined(_DEBUG) && defined(_MSC_VER)
+// #if defined(_DEBUG) && defined(_MSC_VER)
+// doesn't work for my laptop (sergey)
+#if 0
   // When debugging on windows, avoid the obnoxious dialog and make
   // it possible to continue past a LOG(FATAL) in the debugger
   _asm int 3
diff --git a/src/third_party/glog/src/raw_logging.cc b/src/third_party/glog/src/raw_logging.cc
index 50c6a71..b179a1e 100644
--- a/src/third_party/glog/src/raw_logging.cc
+++ b/src/third_party/glog/src/raw_logging.cc
@@ -42,8 +42,8 @@
 #include <fcntl.h>                 // for open()
 #include <time.h>
 #include "config.h"
-#include "glog/logging.h"          // To pick up flag settings etc.
-#include "glog/raw_logging.h"
+#include <glog/logging.h>          // To pick up flag settings etc.
+#include <glog/raw_logging.h>
 #include "base/commandlineflags.h"
 
 #ifdef HAVE_STACKTRACE
diff --git a/src/third_party/glog/src/utilities.h b/src/third_party/glog/src/utilities.h
index c4ae256..5c841a0 100644
--- a/src/third_party/glog/src/utilities.h
+++ b/src/third_party/glog/src/utilities.h
@@ -79,7 +79,7 @@
 #endif
 
 #include "config.h"
-#include "glog/logging.h"
+#include <glog/logging.h>
 
 // There are three different ways we can try to get the stack trace:
 //
diff --git a/src/third_party/glog/src/vlog_is_on.cc b/src/third_party/glog/src/vlog_is_on.cc
index ee0e412..ed88514 100644
--- a/src/third_party/glog/src/vlog_is_on.cc
+++ b/src/third_party/glog/src/vlog_is_on.cc
@@ -40,8 +40,8 @@
 #include <cstdio>
 #include <string>
 #include "base/commandlineflags.h"
-#include "glog/logging.h"
-#include "glog/raw_logging.h"
+#include <glog/logging.h>
+#include <glog/raw_logging.h>
 #include "base/googleinit.h"
 
 // glog doesn't have annotation
diff --git a/src/third_party/glog/src/windows/config.h b/src/third_party/glog/src/windows/config.h
index 114762e..682a1b9 100755
--- a/src/third_party/glog/src/windows/config.h
+++ b/src/third_party/glog/src/windows/config.h
@@ -19,7 +19,7 @@
 #undef HAVE_LIBUNWIND_H
 
 /* define if you have google gflags library */
-#undef HAVE_LIB_GFLAGS
+#define HAVE_LIB_GFLAGS 1
 
 /* define if you have libunwind */
 #undef HAVE_LIB_UNWIND
diff --git a/src/third_party/glog/src/windows/glog/logging.h b/src/third_party/glog/src/windows/glog/logging.h
index 4257375..2f41681 100755
--- a/src/third_party/glog/src/windows/glog/logging.h
+++ b/src/third_party/glog/src/windows/glog/logging.h
@@ -82,8 +82,8 @@
 #include <inttypes.h>           // a third place for uint16_t or u_int16_t
 #endif
 
-#if 0
-#include <gflags/gflags.h>
+#if 1
+#include "third_party/gflags/gflags.h"
 #endif
 
 #ifdef __MINGW32__
