diff --git a/src/third_party/glog/src/config_mac.h b/src/third_party/glog/src/config_mac.h
index a45575b..5f953d1 100644
--- a/src/third_party/glog/src/config_mac.h
+++ b/src/third_party/glog/src/config_mac.h
@@ -131,7 +131,7 @@
 #define PACKAGE_VERSION "0.3.1"
 
 /* How to access the PC from a struct ucontext */
-#define PC_FROM_UCONTEXT uc_mcontext->__ss.__rip
+#undef PC_FROM_UCONTEXT
 
 /* Define to necessary symbol if this constant uses a non-standard name on
    your system. */
