diff --git a/src/third_party/fast/fast.h b/src/third_party/fast/fast.h
index 2b3825a..06fa90e 100644
--- a/src/third_party/fast/fast.h
+++ b/src/third_party/fast/fast.h
@@ -1,6 +1,10 @@
 #ifndef FAST_H
 #define FAST_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef struct { int x, y; } xy; 
 typedef unsigned char byte;
 
@@ -28,4 +32,8 @@ xy* fast12_detect_nonmax(const byte* im, int xsize, int ysize, int stride, int b
 xy* nonmax_suppression(const xy* corners, const int* scores, int num_corners, int* ret_num_nonmax);
 
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
