diff --git a/src/libmv/simple_pipeline/pipeline.cc b/src/libmv/simple_pipeline/pipeline.cc
index 652d70c..a60ba40 100644
--- a/src/libmv/simple_pipeline/pipeline.cc
+++ b/src/libmv/simple_pipeline/pipeline.cc
@@ -42,7 +42,7 @@ void CompleteReconstruction(const Tracks &tracks,
   while (num_resects != 0 || num_intersects != 0) {
     // Do all possible intersections.
     num_intersects = 0;
-    for (int track = 0; track < max_track; ++track) {
+    for (int track = 0; track <= max_track; ++track) {
       if (reconstruction->PointForTrack(track)) {
         LG << "Skipping point: " << track;
         continue;
