diff --git a/src/libmv/simple_pipeline/pipeline.cc b/src/libmv/simple_pipeline/pipeline.cc
index 652d70c..25cd2c2 100644
--- a/src/libmv/simple_pipeline/pipeline.cc
+++ b/src/libmv/simple_pipeline/pipeline.cc
@@ -28,6 +28,10 @@
 #include "libmv/simple_pipeline/tracks.h"
 #include "libmv/simple_pipeline/camera_intrinsics.h"
 
+#ifdef _MSC_VER
+#  define snprintf _snprintf
+#endif
+
 namespace libmv {
 
 void CompleteReconstruction(const Tracks &tracks,
