diff --git a/src/libmv/simple_pipeline/bundle.cc b/src/libmv/simple_pipeline/bundle.cc
index 310660d..f819603 100644
--- a/src/libmv/simple_pipeline/bundle.cc
+++ b/src/libmv/simple_pipeline/bundle.cc
@@ -141,7 +141,6 @@ void Bundle(const Tracks &tracks, Reconstruction *reconstruction) {
   v3d_distortion.p2 = 0;
 
   // Finally, run the bundle adjustment.
-  V3D::optimizerVerbosenessLevel = 1;
   double const inlierThreshold = 500000.0;
   V3D::CommonInternalsMetricBundleOptimizer opt(V3D::FULL_BUNDLE_METRIC,
                                                 inlierThreshold,
