Index: LinearMath/btSerializer.h
===================================================================
--- LinearMath/btSerializer.h	(revision 45919)
+++ LinearMath/btSerializer.h	(working copy)
@@ -25,8 +25,15 @@
 #endif
 #include <string.h>
 
+#if defined(_WIN64)
+#  ifdef __MINGW64__
+#    include <basetsd.h>
+#  endif
+typedef __int64 int_ptr;
+#else
+typedef long int_ptr;
+#endif
 
-
 ///only the 32bit versions for now
 extern unsigned char sBulletDNAstr[];
 extern int sBulletDNAlen;
@@ -247,7 +254,7 @@
 				cp++;
 			}
 			{
-				nr= (long)cp;
+				nr= (int_ptr)cp;
 			//	long mask=3;
 				nr= ((nr+3)&~3)-nr;
 				while (nr--)
@@ -282,7 +289,7 @@
 			}
 
 		{
-				nr= (long)cp;
+				nr= (int_ptr)cp;
 			//	long mask=3;
 				nr= ((nr+3)&~3)-nr;
 				while (nr--)
